src/functions/products/getProductsById/handler.ts

```diff
@@ -0,0 +1,34 @@
+import { formatJSONResponse } from '@libs/api-gateway';
+import { middyfy } from '@libs/lambda';
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+import { products } from 'src/mocks/data';
+import { Product } from 'src/models/Product';
+
+const getProductsById = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  const productId = event.pathParameters?.productId;
+
+  if (productId) {
+    const product = products.find((p) => p.id === productId);
+    if (product) {
+      return apiResponses._200(product);
+    }
+  }
+
+  return apiResponses._400({ message: 'Product not found' });
+};
+
+const apiResponses = {
+  _200: (data: Product) => {
+    return formatJSONResponse({
+      data,
```

Comment:
As per _task 3.1, 3. The response from the lambda should be 1 searched product._ -  In your case its returning an object with 'data' as key and value as the searched object. I believe you can change this to return the correct type as specified in the task. Also, you could place the apiResponses function in an utils folder.

Reply:
The tasks says what data should be sent, but never states how the data should be sent. 
I don't think this should be an issue, as I am the one handling the requests, so I can define the contract over the UI app.